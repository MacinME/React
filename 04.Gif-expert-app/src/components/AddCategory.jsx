import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputValue = (evt) => {
        setInputValue(evt.target.value);
    }

    const onSubmit = (evt) => {
        evt.preventDefault();
        if(inputValue.trim().length <= 1 ) return;

        onNewCategory( inputValue );
        // setCategories((values) => [...values, inputValue]);
        setInputValue('')
    }

  return (
    <form onSubmit={ onSubmit } aria-label='form'>
        <input 
            type="text" 
            placeholder="Search Gifs"    
            value={ inputValue }
            onChange={ onInputValue }
        />
    </form>
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}

