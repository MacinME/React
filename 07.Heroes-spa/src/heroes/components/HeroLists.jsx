import { useMemo } from 'react';
import { getHeroesByPublisher } from '../helpers';
import { HeroCard } from './';

export const HeroLists = ({ publisher }) => {

    const getHeroes = useMemo(() => getHeroesByPublisher( publisher ), [ publisher ]);

  return (

    <div className='row rows-cols-1 row-cols-md-3 g-3'>
        {
            getHeroes.map(hero => ( 
                <HeroCard  
                    key={ hero.id }
                    {...hero}
                />    
            ))
        }
    </div>

  )
}
