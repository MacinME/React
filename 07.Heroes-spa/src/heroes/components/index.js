export * from './HeroLists';
export * from './HeroCard';