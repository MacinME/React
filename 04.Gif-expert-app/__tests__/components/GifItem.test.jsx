import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem';

describe('Test on <GifItem />', () => { 

    const title = "One Punch";
    const image = "https://onepunch/saitama.png";

    test('Must match with snapshot', () => { 
        const { container } = render(
            <GifItem title={ title} image={ image } />
        )    
        
        expect( container ).toMatchSnapshot();
    })

    test('Must display the image with indicated url', () => {
        render(
            <GifItem title={ title } image={ image } />
        );

        // expect(screen.getByRole('img').src ).toBe( image );
        // expect( screen.getByRole('img').alt ).toBe( title );

        const { src, alt} = screen.getByRole('img');
        expect( src ).toBe( image );
        expect( alt ).toBe( title );
    })

    test('Must display the title in the component', () => {
        render(
            <GifItem title={ title } image={ image } />
        );

        expect( screen.getByText( title ) ).toBeTruthy();
    })
 })