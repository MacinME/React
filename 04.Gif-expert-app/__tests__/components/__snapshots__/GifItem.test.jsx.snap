// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifItem /> Must match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://onepunch/saitama.png"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
